### DISCUSSION TITLE
Undefined to an empty string on formState changes

### DISCUSSION THREAD

**Discussion by k-one-o-two** [react-hook-form/react-hook-form#7826] [Q&A]
I have a form and some default values, like this:
```
const defaultValues = {text: 'text'};
reset(defaultValues);
```
In the form, there are two text inputs with names `text` and `title`.

The problem is, the value of `title` is changing into the empty string on every change of form state:
```
  useEffect(() => {
    const form = getValues();
    console.info(form, formState.touchedFields);
  }, [formState]);
```
Gives: `{text: 'text', title: ''}` and an empty object for `touchedFields`. This is quite critical since in the application there's a big difference between an undefined value and an empty string.

How can I make RHF not to change untouched values?

Isolated example: https://codesandbox.io/s/clever-jang-d2s4n?file=/src/App.js

**Comment by bluebill1049** [comment]
native input will not return `undefined`, and this library doesn't support `undefined` as well

**Reply by k-one-o-two** [reply]
Hm, could you suggest any workaround?

**Reply by Moshyfawn** [reply]
There's no workaround as you're using an uncontrolled component, which means the source of truth for your value is in the input. Native web inputs don't return `undefined`, only empty strings in case of an empty value.
My personal take on this: I'd suggest to not go against the web standards.

You only option is to convert your input to controlled and use `Controller` / `useController` with your custom logic to handle `undefined` vs `empty string`.
